tests/extmod/vfs_posix_ilistdir_filter.py

```diff
@@ -0,0 +1,45 @@
+# Test ilistdir filter of . and .. for VfsPosix.
+
+try:
+    import os
+
+    os.VfsPosix
+except (ImportError, AttributeError):
+    print("SKIP")
+    raise SystemExit
+
+
+def test(testdir):
+    vfs = os.VfsPosix(testdir)
+
+    expected = {".a": True, "..a": True, "...a": True, "a.": True}
+    actual = {}
+
+    for dir in expected:
+        vfs.mkdir(dir)
+
+    for entry in vfs.ilistdir("/"):
+        actual[entry[0]] = True
+
+    assert expected == actual
+
+
+# We need an empty directory for testing.
+# Skip the test if it already exists.
+temp_dir = "vfs_posix_ilistdir_filter_test_dir"
+try:
+    os.stat(temp_dir)
+    print("SKIP")
+    raise SystemExit
+except OSError:
+    pass
+
+os.mkdir(temp_dir)
+
+test(temp_dir)
+
+# Remove tempdir.
+for td in os.listdir(temp_dir):
+    os.rmdir("/".join((temp_dir, td)))
+
+os.rmdir(temp_dir)
```

Comment:
Should probably put this in a `finally:` to have the best change of removal.

Reply:
Just so I understand your comment, are you suggesting to change the code to look more like:

```
try:
    os.mkdir(temp_dir)
    test(temp_dir)
finally:
    # Remove tempdir.
    for td in os.listdir(temp_dir):
        os.rmdir("/".join((temp_dir, td)))
    os.rmdir(temp_dir)
```

Reply:
the mkdir should probably be before the try: but yes, that is the basic idea